src/cata_assert.h

```diff
@@ -0,0 +1,27 @@
+// NOLINTNEXTLINE(cata-header-guard)
+// Due to an inability to supress assert popups when building against mingw-w64 and running on wine
+// We are wrapping the assert macro so that we can substitute functional behavior with that setup
+
+// This copies the semantics of cassert, re-including the file re-defines the macro.
+#undef cata_assert
+
+// Might as well handle NDEBUG at the top level instead of just wrapping one variant.
+#ifdef NDEBUG
+#define cata_assert(_Expression) ((void)0)
```

Comment:
```suggestion
#define cata_assert(expression) ((void)0)
```
Identifiers starting with an underscore are reserved for the standard library.